templates/types/streaming/nextjs/app/api/chat/route.ts

```diff
@@ -76,6 +76,20 @@ export async function POST(request: NextRequest) {
         `Retrieved ${nodes.length} sources to use as context for the query`,
       );
     });
+    callbackManager.on("llm-tool-call", (event) => {
+      const { name, input } = event.detail.payload.toolCall;
+      const inputString = Object.entries(input)
+        .map(([key, value]) => `${key}: ${value}`)
+        .join(", ");
+      appendEventData(
+        vercelStreamData,
+        `Using tool: '${name}' with inputs: '${inputString}'`,
+      );
+    });
+    callbackManager.on("llm-tool-result", (event) => {
+      const { toolCall, toolResult } = event.detail.payload;
+      appendToolData(vercelStreamData, toolCall, toolResult);
+    });
```

Comment:
move this stuff into a `createCallbackManager` function returning the `CallbackManager`